File: darts-java/src/main/java/com/stride/bed/Double.java
package com.stride.bed;

public class Double implements Scorable {
    public static final int MULTIPLIER = 2;

    private int score;

    public Double(int bedValue) {
        if (bedValue < 1 || bedValue > 20) {
            throw new IllegalArgumentException("Throw bedScore must be in range 1..20");
        }
        this.score = bedValue * MULTIPLIER;
    }

    public int bedScore() {
        return score;
    }
}

File: darts-ruby/lib/darts/models/throw/double.rb
module Throw
  class Double
    MULTIPLIER = 2

    attr_reader :bed_score

    def initialize(bed_value)
      raise ArgumentError.new('Throw must be in range 1..20') if bed_value < 1 || bed_value > 20
      @bed_score = bed_value * MULTIPLIER
    end
  end
end